src/core/core.scale.js

```diff
@@ -60,6 +60,111 @@ module.exports = function(Chart) {
 			context.measureText(tick).width;
 	}
 
+	/**
+	 * Calculate the mid point between 2 values
+	 *
+	 * This is really the same thing as the average
+	 *
+	 * @param point1 numeric value
+	 * @param point2 numeric value
+	 * @return the mid point between point1 and point2
+	 */
+	function midPoint(point1, point2) {
+		return (point1 + point2) / 2;
+		// return point1 + ((point2 - point1) / 2);
+	}
+
+	/**
+	 * Calculate text origin and alignment
+	 *
+	 * Center origin and alignment prefered, but shift to start or end when using
+	 * center would cause truncation.
+	 *
+	 * @param isForward true when the text is to be placed right side up
+	 * @param centerPoint coordinate value to center text in the target area
+	 * @param halfWidth one half the width of the text to be placed
+	 * @param maxExtend the maximum coordinate beyond which truncation will occur
+	 * @return object with text placment origin coordinate and alignment
+	 */
+	function useEndOnTruncate(isForward, centerPoint, halfWidth, maxExtent) {
+		var positionProperties = {
```

Comment:
I like your way.  I knew I wanted to return an object when I started, so the first thing I did was create it, to be populated and returned.  Creating the object only as it is being returned is cleaner.